cli/src/main/java/com/devonfw/tools/ide/property/Property.java

```diff
@@ -449,21 +452,24 @@ public boolean matches(String nameOrAlias) {
 
   /**
    * @return {@code true} if this {@link Property} is valid, {@code false} if it is {@link #isRequired() required} but no {@link #getValue() value} has been
-   * set.
-   * @throws RuntimeException if the {@link #getValue() value} is violating given constraints. This is checked by the optional {@link Consumer} function given
-   * at construction time.
+   *     set.
+   * @throws RuntimeException if the {@link #getValue() value} is violating given constraints. This is checked by the optional {@link Consumer} function
+   *     given at construction time.
    */
-  public boolean validate() {
+  public ValidationResult validate() {
+
+    ValidationState state = new ValidationState(this.getNameOrAlias());
 
     if (this.required && (getValue() == null)) {
-      return false;
+      state.addErrorMessage("Property " + this.name + " is required but no value has been set");
```

Comment:
The name of the property is now already included by the `ValidationState`:
```suggestion
      state.addErrorMessage("Value is required and cannot be empty.");
```